registry/registry.go

```diff
@@ -741,20 +833,247 @@ func downloadVersion(opts *VersionOptions) (ver *Version, attachments []*kivik.A
 		}
 	}
 
+	// Reader for filesize
 	counter := &Counter{}
 	var reader io.Reader = buf
 	reader = io.TeeReader(reader, counter)
 
-	var packVersion string
+	// Reader for tarball content
+	var fileContent = new(bytes.Buffer)
+	reader = io.TeeReader(reader, fileContent)
+
+	// Reading the tarball content
+	tarball, err := ReadTarballVersion(reader, contentType, url)
+	tarball.ContentType = contentType
+
+	// Retreiving the tarball manifest
+	parsedManifest := tarball.Manifest
+
+	if !tarball.HasPrefix {
+		tarPrefix = ""
+	}
+
+	// Checks
+	if _, erre := tarball.CheckEditor(); erre != nil {
+		err = multierror.Append(err, erre)
+	}
+	if _, errs := tarball.CheckSlug(); errs != nil {
+		err = multierror.Append(err, errs)
+	}
+	if _, errv := tarball.CheckVersion(opts.Version); errv != nil {
+		err = multierror.Append(err, errv)
+	}
+
+	// Handling assets
+	attachments, err = HandleAssets(tarball, opts, buf, url, attachments)
```

Comment:
You are overwriting `err` here, and will lose the potential errors from `CheckEditor`, `CheckSlug` and `CheckVersion`